requirements/runtime.in

```diff
@@ -1,17 +1,9 @@
 -c runtime-constraints.in  # limits known broken versions
 
-# NOTE: v6.1 is needed to support metadata v2.4 including PEP 639
+# NOTE: v6.1 is needed to support metadata v2.4 including PEP 639,
+# and also uploading via Trusted Publishing
 twine >= 6.1
 
-# NOTE: Used to detect an ambient OIDC credential for OIDC publishing,
-# NOTE: as well as PEP 740 attestations.
-id ~= 1.0
```

Comment:
@facutuesca upon rebase, you'll probably need to remove the type stub that #381 added and drop this dep from the pre-commit config.